MachObfuscator/Commons/Concurrent.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Concurrent.swift
+//  MachObfuscator
+//
+
+import Foundation
+
+extension Array {
+    // Building block for concurrent operations
+    // This implementation is best for realativly small arrays with costly per-item computation
+    //
+    // Insipired by https://talk.objc.io/episodes/S01E90-concurrent-map
+    fileprivate func concurrentMap_impl<B>(_ transform: @escaping (Element) -> B?) -> [B?] {
+        var result = [B?](repeating: nil, count: count)
+        let q = DispatchQueue(label: "sync queue")
+        DispatchQueue.concurrentPerform(iterations: count) { idx in
+            let element = self[idx]
+            let transformed = transform(element)
+            q.sync {
+                result[idx] = transformed
+            }
+        }
+        return result
+    }
+
+    func concurrentMap<B>(_ transform: @escaping (Element) -> B) -> [B] {
+        return concurrentMap_impl(transform).map { $0! }
+    }
+
+    func concurrentCompactMap<B>(_ transform: @escaping (Element) -> B?) -> [B] {
+        return concurrentMap_impl(transform).compactMap { $0 }
+    }
+
+    func concurrentCompactMap_improbable<B>(numPartitions partitions: Int, _ transform: @escaping (Element) -> B?) -> [B] {
```

Comment:
Is this method needed/used at all?

Reply:
currently it is not used. I tried it in a few places (mainly processing cstrings) but it did not bring substantial improvements. I left it, because it may be useful somewhere.